letsencrypt/crypto_util.py

```diff
@@ -18,12 +18,16 @@
 from letsencrypt import interfaces
 from letsencrypt import le_util
 
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import ec
 
 logger = logging.getLogger(__name__)
 
 
 # High level functions
-def init_save_key(key_size, key_dir, keyname="key-letsencrypt.pem"):
+def init_save_key(key_algo, rsa_key_size, ecdsa_curve, key_dir, keyname="key-letsencrypt.pem"):
```

Comment:
This function now has a weird signature, in that you're required to specify `rsa_key_size` even if you're creating an ECDSA key. It probably makes more sense to change this function to `save_key(key_pem, key_dir...)` and have the code in client.py call `make_key_rsa` or `make_key_ecdsa` depending on the config, and then call `save_key` on the result.